quickwit/quickwit-storage/src/local_file_storage.rs

```diff
@@ -175,13 +168,18 @@ impl Storage for LocalFileStorage {
 
     async fn delete(&self, path: &Path) -> StorageResult<()> {
         let full_path = self.root.join(path);
-        missing_file_is_ok(fs::remove_file(full_path).await)?;
-        let parent = path.parent();
-        if parent.is_none() {
-            return Ok(());
+        ignore_error_kind!(ErrorKind::NotFound, fs::remove_file(full_path).await)?;
+        if let Some(parent) = path.parent() {
+            if let Err(error) = delete_all_dirs(&self.root, parent).await {
+                warn!(error=?error, path=%path.display(), "Failed to delete directory.");
+            }
         }
-        if let Err(error) = delete_all_dirs(self.root.to_path_buf(), parent.unwrap()).await {
-            warn!(error = ?error, path = %path.display(), "Failed to clean up parent directories.");
+        Ok(())
+    }
+
+    async fn batch_delete<'a>(&self, paths: &[&'a Path]) -> StorageResult<()> {
+        for path in paths {
+            self.delete(path).await?;
```

Comment:
Awaiting in this loop is blocking. The futures will run sequentially as opposed to running concurrently. I think we want to run them concurrently right? `future::join_all` or `stream.buffer_unordered(n)`